sdk/storage/azblob/README.md

```diff
@@ -0,0 +1,362 @@
+
+### Install the package
+* Install the Azure blob storage for Go with `go get`:
+  ```bash
+  go get github.com/Azure/azure-sdk-for-go/sdk/storage/azblob
+  ```
+  
+#### Create the client
+
+`azblob` allows you to interact with three types of resources :-
+
+* [Azure storage accounts][azure_storage_account].
+* [Containers](https://azure.microsoft.com/en-in/overview/what-is-a-container/#overview) within those storage accounts.
+* [Blobs](https://azure.microsoft.com/en-in/services/storage/blobs/#overview) (block blobs/ page blobs/ append blobs) within those containers.
+
+Interaction with these resources starts with an instance of a [client](#clients). 
+To create a client object, you will need the account's blob service endpoint URL and a credential that allows you to access the account. 
+The `endpoint` can be found on the page for your storage account in the [Azure Portal][azure_portal_account_url] under the "Access Keys" section or by running the following Azure CLI command:
+
+```bash
+# Get the table service URL for the account
+az storage account show -n mystorageaccount -g MyResourceGroup --query "primaryEndpoints.table"
+```
+
+Once you have the account URL, it can be used to create the service client:
+```golang
+cred, err := azblob.NewSharedKeyCredential("myAccountName", "myAccountKey")
+handle(err)
+serviceClient, err := azblob.NewServiceClient("https://<myAccountName>.blob.core.windows.net/", cred, nil)
+handle(err)
+```
+
+For more information about blob service URL's and how to configure custom domain names for Azure Storage check out the [official documentation][azure_portal_account_url]
+
+#### Types of credentials
+
+The clients support different forms of authentication. Storage accounts can use a Shared Key Credential, Connection
+String, or a Shared Access Signature Token for authentication
+in [`azidentity`](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity)
+like `azidentity.NewDefaultAzureCredential()`.
+
+The azblob package supports any of the types that implement the `azcore.TokenCredential` interface, authorization via a
+Connection String, or authorization with a Shared Access Signature Token.
+
+##### 1. Creating the client from a shared key
+
+To use an account [shared key][azure_shared_key] (aka account key or access key), provide the key as a string. This can
+be found in your storage account in the [Azure Portal][azure_portal_account_url] under the "Access Keys" section or by
+running the following Azure CLI command:
+
+```bash
+az storage account keys list -g MyResourceGroup -n MyStorageAccount
+```
+
+Use AAD authentication as the credential parameter to authenticate the client:
+```golang
+cred, err := azidentity.NewDefaultAzureCredential(nil)
+handle(err)
+serviceClient, err := aztables.NewServiceClient("https://<myAccountName>.table.core.windows.net/", cred, nil)
+handle(err)
```

Comment:
This sample is for shared key and you're showing DefaultAzureCredential

Reply:
Ahh my bad. This is taking more time than usual. Thanks for pointing out.